### DISCUSSION TITLE
Pre/Post build and Pre/Post upload steps

### DISCUSSION THREAD

**Discussion by brailovskii** [stm32duino/Arduino_Core_STM32#1762] [Ideas]
I think it would be useful feature. Most of the IDEs have these options.
I wrote some usage ideas in parentheses, but these may be extended.

Pre build step ==> run user defined script before building sketch
Post build step ==> run user defined script after building sketch(maybe converting generated binary file to HEX, or encrypting or smth. Else)

Pre upload step ==> run user defined script before uploading to board(initialize board, set it to boot loader mode; or even use user defined uploader)
Post upload step ==> run user defined script after uploading to board(prepare board to exit from boot loader mode)

**Comment by PerennialNovice** [comment]
you can have this (and a lot more) with [sloeber](https://eclipse.baeyens.it/)
but, to be honest, I am not sure about the pre/post upload steps...

**Comment by fpistm** [comment]
Hi @brailovskii 
You can already do this as Arduino IDE provides a way to (re-)define the platform.txt by creating a platform.local.txt:
https://arduino.github.io/arduino-cli/0.21/platform-specification/#platformlocaltxt

In this file simply define your own pre/post target:
https://arduino.github.io/arduino-cli/0.21/platform-specification/#pre-and-post-build-hooks-since-arduino-ide-165
